### PR TITLE
Some work on the OS400 port

### PR DESCRIPTION
**Pull request by monnerat**
Thanks to pub400.com, I have finally been able to do some work for this platform.
Mainly some fixes, make it up to date, support blob-, webosockets- and header APIs.
Please see the commit comments for more information.

Note that nobody ever used the mime api from ILE/RPG since its creation: I've fixed errors that would have prevented any correct functioning!
 
I've also considered a possible work on gskit, but the system cert/key files are restricted and there's no way to create user alternative ones. I thus may update code and make it compile successfully, but I can't run it :-( I left a message to request permissions but it seems the admin is ill since febuary. BTW: what would you require to save GsKit? TLS 1.3 I guess... anything more ?

### CODE REVIEW

File: packages/OS400/makefile.sh [hunk 1]
```diff
@@ -66,7 +66,37 @@ do      MEMBER="`basename \"${TEXT}\" .OS400`"
...
+then    CMD="CRTSRCPF FILE(${TARGETLIB}/RPGXAMPLES) RCDLEN(240)"
+        CMD="${CMD} CCSID(${TGTCCSID}) TEXT('ILE/RPG examples')"
+        CLcommand "${CMD}"
+fi
+
+
+#       Copy RPG examples if needed.
+
+for EXAMPLE in "${SCRIPTDIR}/rpg-examples"/*
+do      MEMBER="`basename \"${EXAMPLE}\"`"
```
[the remark this answers is not in the document]
Comment (monnerat) [reply packages/OS400/makefile.sh:86]: Sonatype is VERY bad confusing an IBM i Qshell script for a bash one !

**Comment by monnerat** [comment]
Turning it to draft: still some details to tune!

File: packages/OS400/make-lib.sh [hunk 2]
```diff
@@ -70,23 +70,23 @@ sed -e ':begin'                                                         \
...
-    fi
-  fi
+if action_needed "${LIBIFSNAME}/CHKSTRINGS.PGM" "${SCRIPTDIR}/chkstrings.c"
+then    CMD="CRTBNDC PGM(${TARGETLIB}/CHKSTRINGS)"
+        CMD="${CMD} SRCSTMF('${SCRIPTDIR}/chkstrings.c')"
+        CMD="${CMD} INCDIR('${TOPDIR}/include/curl' '${TOPDIR}/include'"
+        CMD="${CMD} '${SRCDIR}' ${INCLUDES})"
+        CMD="${CMD} TGTCCSID(${TGTCCSID})"
+        CLcommand -i "${CMD}"
+        if [ "${?}" -ne 0 ]
```
[the remark this answers is not in the document]
Comment (monnerat) [reply packages/OS400/make-lib.sh]: Done

**Comment by jonrumsey** [comment]
Thanks @monnerat - the changes look good.

**Comment by jay** [comment]
Thanks. I left the commits separate and referenced this PR in each one of them.

**Comment by monnerat** [comment]
Thanks for merge @jay !

**Pull request closed without merging**